src/components/structures/MatrixChat.js

```diff
@@ -491,39 +488,45 @@ module.exports = React.createClass({
 
     // switch view to the given room
     //
-    // eventId is optional and will cause a switch to the context of that
-    // particular event.
-    // @param {Object} thirdPartyInvite Object containing data about the third party
+    // @param {Object} room_info Object containing data about the room to be joined
+    // @param {string} room_info.room_id ID of the room to join. One of room_id or room_alias must be given.
+    // @param {string} room_info.room_alias Alias of the room to join. One of room_id or room_alias must be given.
+    // @param {boolean} room_info.auto_join If true, automatically attempt to join the room if not already a member.
+    // @param {string} room_info.show_settings ??
+    // @param {string} room_info.event_id ID of the event in this room to show: this will cause a switch to the
+    //                                    context of that particular event. Optional.
```

Comment:
there are ways of marking it optional in the jsdoc (notably sticking an '=' after the type): http://usejsdoc.org/tags-param.html#optional-parameters-and-default-values